modules/ListView.js

```diff
@@ -367,48 +417,110 @@
       return model ? this.getChildView(this.__modelToViewMap[model[this.__modelId]]) : undefined;
     },
 
+    /**
+     * @return {Boolean} returns true if there exists any generated item views
+     * @method hasItemViews
+     */
+    hasItemViews: function() {
+      return !_.isEmpty(this.getItemViews());
+    },
+
+    /**
+     * @return {Array of views} Returns unordered list of views generated by this list view
+     * @method getItemViews
+     */
+    getItemViews: function() {
+      var view = this;
+      var orderedViewIds = _.map(this.__orderedModelIdList, this.__getViewIdFromModelId, this);
+      return _.map(orderedViewIds, this.getChildView, this);
+    },
+
     /************** Private methods **************/
 
     /**
-     * Creates a new child view if there doesn't exist one for a model
-     * @method __createChildViews
+     * Creates all needed item views that don't exist from modelsToRender()
+     * @method __createItemViews
      * @private
+     * @return {Array} each object in array contains a 'view' and 'indexOfModel' field
      */
-    __createChildViews: function() {
-      var newChildViews = [];
-      _.each(this.modelsToRender(), function(model) {
+    __createItemViews: function() {
+      var newItemViews = [];
+      _.each(this.modelsToRender(), function(model, indexOfModel) {
         var childView = this.getChildViewFromModel(model);
         if (!childView) {
-          newChildViews.push(this.__createChildView(model));
+          newItemViews.push({ view: this.__createItemView(model, true), indexOfModel: indexOfModel });
```

Comment:
I find multiple properties easier to read if they are on separate lines as a block.  Easier to tell when one ends and the other starts, esp when there are method calls with parameters (extra commas).